src/keymap-manager.coffee

```diff
@@ -214,40 +214,51 @@ class KeymapManager
   Section: Adding and Removing Bindings
   ###
 
-  # Public: Add sets of key bindings grouped by CSS selector.
+  # Extended: Construct [KeyBindings]{KeyBinding} from an object grouping them by CSS selector.
   #
   # * `source` A {String} (usually a path) uniquely identifying the given bindings
   #   so they can be removed later.
   # * `bindings` An {Object} whose top-level keys point at sub-objects mapping
   #   keystroke patterns to commands.
   # * `priority` A {Number} used to sort keybindings which have the same
   #   specificity. Defaults to `0`.
-  add: (source, keyBindingsBySelector, priority=0, throwOnInvalidSelector=true) ->
-    addedKeyBindings = []
+  build: (source, keyBindingsBySelector, priority=0, throwOnInvalidSelector=true) ->
+    bindings = []
     for selector, keyBindings of keyBindingsBySelector
       # Verify selector is valid before registering any bindings
-      if throwOnInvalidSelector and not isSelectorValid(selector.replace(/!important/g, ''))
-        console.warn("Encountered an invalid selector adding key bindings from '#{source}': '#{selector}'")
-        return
+      unless isSelectorValid(selector.replace(/!important/g, ''))
+        if throwOnInvalidSelector
```

Comment:
Oh, whoops. Looking back through the history, `throwOnInvalidSelector` is intended to prevent an access to `document` during snapshotting. By rearranging the conditional here I've broken that. Follow-up PR incoming :rocket: